api/client/volume/client_volume_test.go

```diff
@@ -29,3 +28,144 @@ func TestClientTLS(t *testing.T) {
 
 	require.NoError(t, err)
 }
+
+func TestClientCredCreate(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
If i understand correctly, this test code starts its own server, returns its own responses and bypasses the OSD code all together. Don't see why is it needed .. ?

Reply:
This tests the client code alone. Mock driver for volumeDriver would support end-to-end ( with the volume driver implementation being mocked).Still under development.